internal/restic/snapshot_find.go

```diff
@@ -12,13 +12,32 @@ import (
 // ErrNoSnapshotFound is returned when no snapshot for the given criteria could be found.
 var ErrNoSnapshotFound = errors.New("no snapshot found")
 
-// findLatestSnapshot finds latest snapshot with optional target/directory, tags, hostname, and timestamp filters.
-func findLatestSnapshot(ctx context.Context, be Lister, loader LoaderUnpacked, hosts []string,
-	tags []TagList, paths []string, timeStampLimit *time.Time) (*Snapshot, error) {
+// A SnapshotFilter denotes a set of snapshots based on hosts, tags and paths.
+type SnapshotFilter struct {
```

Comment:
(Just a question, not a request to change the code) I'm curious why `FindLatest` and `FindAll` are methods of `SnapshotFilter` instead of passing a `SnapshotFilter` option struct to those methods. That is what is the benefit compared to a signature like `func FindLatest(ctx context.Context, be Lister, loader LoaderUnpacked, snapshotID string, opts *SnapshotFilter) (*Snapshot, error)`?

Reply:
Mainly because FindLatest and FindAll are too general as names for snapshot finding functions in package restic, but I found the old names too long and not descriptive enough. In other words, the type is used as a namespacing device.

Reply:
Thanks for the explanation.